GoMeet/src/main/webapp/test/spec/spec-vote-meeting-time.js

```diff
@@ -0,0 +1,316 @@
+    }
+  });
+
+  it('creates rows in the table with id="meeting-times-table" with the data received \
+      disabling the vote buttons of times that the user has voted for', async function() {
+    let votedTimesTherese = new Set(['def456', 'ghi789']);
+    generateVoteTimeForm(MEETING_TIME_DATA, "therese@test.com", votedTimesTherese);
+    verifyRowData();
+    let table = document.getElementById('meeting-times-table');
+    for (let i = 1; i < table.rows.length; i++) {
+      let dataRow = table.rows.item(i);
+      let voteButton = dataRow.children.item(3).children.item(0);
+      // If the button if associated with a time not voted for yet, 
+      // button should not be disabled.
+      if (!votedTimesTherese.has(voteButton.id)) {
+        expect(voteButton.disabled).toBe(false); 
+      } else {
+        expect(voteButton.disabled).toBe(true); 
+      }
+    }
+  });
+
+  it('creates rows in the table with id="meeting-times-table" with the data received \
+      enabling all vote buttons if user has not voted yet', async function() {
+    let votedTimesNewUser = new Set();
+    generateVoteTimeForm(MEETING_TIME_DATA, "newuser@test.com", votedTimesNewUser);
+    verifyRowData();
+    let table = document.getElementById('meeting-times-table');
+    for (let i = 1; i < table.rows.length; i++) {
+      let dataRow = table.rows.item(i);
+      let voteButton = dataRow.children.item(3).children.item(0);
+      expect(voteButton.disabled).toBe(false); // 'currentUser' has not voted at all
+    }
+  });
+  afterEach(function() {
+    // Reset the 'meeting-times-table' table to JUST include the headers.
+    let table = document.getElementById('meeting-times-table');
+    for (let i = table.rows.length - 1; i > 0; i--) {
+      table.deleteRow(i);
+    }
+  });
+});
+
+// TESTS FOR sortTimeByVotes
+describe('sortTimeByVotes', function() {
+  let sortedTimeData;
+  beforeEach(function() {
+    sortedTimeData = [
+      {
+        datetime: '2021-01-27T11:20',
+        voteCount: 3,
+        voters: ['anna@test.com', 'therese@test.com', 'vivian@test.com'],
+      },
+      {
+        datetime: '2021-01-28T11:20',
+        voteCount: 2,
+        voters: ['anna@test.com', 'therese@test.com'],
+      },
+      {
+        datetime: '2021-01-28T17:20',
+        voteCount: 1,
+        voters: ['anna@test.com'],
+      }
+    ];
+  });
+
+  it('sorts the timeData objects by voteCount in decreasing order', function() {
+    let timeData = [
+      {
+        datetime: '2021-01-28T17:20',
+        voteCount: 1,
+        voters: ['anna@test.com'],
+      },
+      {
+        datetime: '2021-01-28T11:20',
+        voteCount: 2,
+        voters: ['anna@test.com', 'therese@test.com'],
+      },
+      {
+        datetime: '2021-01-27T11:20',
+        voteCount: 3,
+        voters: ['anna@test.com', 'therese@test.com', 'vivian@test.com'],
+      }
+    ];
+    sortTimesbyVotes(timeData);
+    expect(timeData).toEqual(sortedTimeData);
```

Comment:
optional - could be good to check a bit more thoroughly. Similar to when tests were passing for the servlet tests even though the convertToJson was returning null. Maybe just a check of the size of timeData

Reply:
Yep, I've added a check for the length to be extra careful as you mentioned. 
But should be good in this case since I've already hardcoded the sortedTimeData values in the beforeEach() function, so it's certain that sortedTimeData is not null 👍